### PR TITLE
fix: Only override `config.devtool` in non-production environments

### PR DESCRIPTION
**Pull request by iker-barriocanal**
Changing `config.devtool` in non-production environments is not recommended and [may cause regressions](https://github.com/vercel/next.js/blob/master/errors/improper-devtool.md). It's overridden by Next.js in development environments and a warning is logged in console; so to avoid this warning, `config.devtool` is only changed in non-development environments.

Links to docs (WIP) of customizing the Next.js SDK initialization have also been added.

### CODE REVIEW

**Review by lobsterkatie — APPROVED** [review]
LGTM, but let's wait for the docs to land before merging.

**Comment by rhcarvalho** [comment]
> LGTM, but let's wait for the docs to land before merging.

Link to docs PR?

**Comment by iker-barriocanal** [comment]
> Link to docs PR?

Docs PR: https://github.com/getsentry/sentry-docs/pull/3319

**Comment by iker-barriocanal** [comment]
Since what was pending for docs has been removed, merging the PR.

**Pull request merged**
